lib/Echidna/Exec.hs

```diff
@@ -106,62 +140,65 @@ instance HTraversable VMAction where
 
 
 eCommandUsing :: (MonadGen n, MonadTest m, Typeable a)
-              => (VMAction Concrete -> m a)
+              => n SolCall
+              -> (VMAction Concrete -> m a)
               -> (VM -> Bool)
-              -> [SolSignature]
               -> Command n m VMState
-eCommandUsing o p ts = Command (\_ -> pure $ Call <$> genInteractions ts) o
+eCommandUsing gen ex p = Command (\_ -> pure $ Call <$> gen) ex
   [ Ensure $ \_ (VMState v) _ _ -> assert $ p v
   , Update $ \(VMState v) (Call c) _ -> VMState $ execState (execCall c) v
   ]
   
 
-eCommand :: (MonadGen n, MonadTest m) => (VM -> Bool) -> [SolSignature] -> Command n m VMState
-eCommand = eCommandUsing (\_ -> pure ())
+eCommand :: (MonadGen n, MonadTest m) => n SolCall -> (VM -> Bool) -> Command n m VMState
+eCommand x y = eCommandUsing x (\_ -> pure ()) y
 
 
 eCommandCoverage :: (MonadGen n, MonadTest m, MonadState VM m, MonadReader CoverageRef m, MonadIO m)
-                 => (VM -> Bool) -> [SolSignature] -> Command n m VMState
-eCommandCoverage = eCommandUsing $ \(Call c) -> execCallCoverage c
+                 => [SolCall] -> (VM -> Bool) -> [SolSignature] -> [Command n m VMState]
+eCommandCoverage cov p ts = case cov of
+  [] -> [eCommandUsing (genInteractions ts) (\(Call c) -> execCallCoverage c) p]
+  xs -> map (\x -> eCommandUsing (mutateCall x) (\(Call c) -> execCallCoverage c) p) xs
```

Comment:
If I'm reading this correctly, we'll stop synthesizing new inputs after the first epoch. Can we instead Gen.choice between mutation and generation?